hack/go-mod-prepare.sh

```diff
@@ -1,8 +1,13 @@
 #!/usr/bin/env bash
 
-ROOTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")/.." && pwd)"
+set -e
 
-cat > "${ROOTDIR}/go.mod" << EOF
+SCRIPTDIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+ROOTDIR="$(git -C "$SCRIPTDIR" rev-parse --show-toplevel)"
+
+set -x
+
+tee "${ROOTDIR}/go.mod" <<EOF
```

Comment:
hack/validate/shfmt is very adamant about putting a space before `EOF`. https://github.com/moby/moby/actions/runs/3498071354/jobs/5858029802
```suggestion
tee "${ROOTDIR}/go.mod" << EOF
```